File: services/api/features/agent/services/embeddingClient.js
const EMBEDDING_URL = (process.env.EMBEDDING_URL || 'http://127.0.0.1:5004').trim().replace(/\/$/, '');
const EMBEDDING_API_KEY = (process.env.EMBEDDING_API_KEY || '').trim();
const EMBED_TIMEOUT_MS = Math.max(3000, parseInt(process.env.AGENT_EMBED_TIMEOUT_MS || '8000', 10) || 8000);

function embedHeaders() {
  const h = { 'Content-Type': 'application/json' };
  if (EMBEDDING_API_KEY) {
    h.Authorization = `Bearer ${EMBEDDING_API_KEY}`;
    h['X-Embedding-Token'] = EMBEDDING_API_KEY;
  }
  return h;
}

/**
 * @param {string} text
 * @returns {Promise<number[] | null>}
 */
async function embedOne(text) {
  const t = String(text || '').trim();
  if (!t || !EMBEDDING_URL) return null;

  const controller = new AbortController();
  const timer = setTimeout(() => controller.abort(), EMBED_TIMEOUT_MS);

  try {
    const res = await fetch(`${EMBEDDING_URL}/embed_one`, {
      method: 'POST',
      headers: embedHeaders(),
      body: JSON.stringify({ text: t.slice(0, 512) }),
      signal: controller.signal,
    });
    if (!res.ok) return null;
    const data = await res.json().catch(() => ({}));
    return Array.isArray(data.embedding) ? data.embedding : null;
  } catch {
    return null;
  } finally {
    clearTimeout(timer);
  }
}

module.exports = { embedOne };

File: services/ai/embedding_client.py
"""HTTP client helpers for the Flag BGE-M3 embedding service."""
from __future__ import annotations

import os

EMBEDDING_URL = os.environ.get("EMBEDDING_URL", "http://localhost:5004").rstrip("/")
EMBEDDING_API_KEY = os.environ.get("EMBEDDING_API_KEY", "").strip()
EMBED_TIMEOUT_SEC = float(os.environ.get("EMBED_TIMEOUT_SEC", "120"))


def embedding_request_headers() -> dict[str, str]:
    headers = {"Content-Type": "application/json"}
    if EMBEDDING_API_KEY:
        headers["Authorization"] = f"Bearer {EMBEDDING_API_KEY}"
        headers["X-Embedding-Token"] = EMBEDDING_API_KEY
    return headers
